java/ingest/ingest-runner/src/main/java/sleeper/ingest/impl/recordbatch/arraylist/ArrayListRecordBatch.java

```diff
@@ -129,29 +131,33 @@ private void flushToLocalDiskAndClear() throws IOException {
         if (inMemoryBatch.isEmpty()) {
             LOGGER.info("There are no records to flush");
         } else {
-            long time1 = System.currentTimeMillis();
+            Instant startTime = Instant.now();
             String outputFileName = String.format("%s/localfile-batch-%s-file-%09d.parquet",
                     localWorkingDirectory,
                     uniqueIdentifier,
                     batchNo);
             inMemoryBatch.sort(new RecordComparator(sleeperSchema));
-            long time2 = System.currentTimeMillis();
+            Instant writeTime = Instant.now();
             // Write the records to a local Parquet file. The try-with-resources block ensures that the writer
             // is closed in both success and failure.
             try (ParquetWriter<Record> parquetWriter = parquetConfiguration.createParquetWriter(outputFileName)) {
                 for (Record record : inMemoryBatch) {
                     parquetWriter.write(record);
                 }
             }
-            long time3 = System.currentTimeMillis();
-            LOGGER.info(String.format("Wrote %d records to local file in %.1fs (%.1f/s) [sorting %.1fs (%.1f/s), writing %.1fs (%.1f/s)] - filename: %s",
+            Instant finishTime = Instant.now();
+            LoggedDuration wholeDuration = LoggedDuration.between(startTime, finishTime);
+            LoggedDuration sortDuration = LoggedDuration.between(startTime, writeTime);
+            LoggedDuration writeDuration = LoggedDuration.between(writeTime, finishTime);
+            LOGGER.info(String.format("Wrote %d records to local file in %ss (%.1f/s) " +
+                            "[sorting %ss (%.1f/s), writing %ss (%.1f/s)] - filename: %s",
```

Comment:
This String.format call seems inconsistent with the rest of the logs. There are other places where the record rates are logged directly to SLF4J. Which way is better?

Reply:
I've replaced the `String.format` call with using the slf4j logger format and a `DecimalFormatter`